benchmark/UnsortedAndSortedJoinBenchmark.cpp

```diff
@@ -0,0 +1,356 @@
+      return std::to_string(object);
+    } else {
+
+      // Converting the vector is a bit more involved.
+      std::stringstream sstream;
+      sstream << "{";
+
+      // Add every number to the stream.
+      std::ranges::for_each(object,
+          [&sstream](size_t number){sstream << number << ", ";}, {});
+
+      // Only return part of the output, because we have a ", " too much.
+      return sstream.str().substr(0, sstream.str().length() - 2) + "}";
+    }
+  };
+
+  // The name of the benchmark table, this function is creating. A bit ugly,
+  // because std::vector<size_t> needs special behaviour and we do not know,
+  // which argument is the vector.
+  std::stringstream tableDescriptor;
+  tableDescriptor << "Benchmarks with " << ((overlap) ? "" : "no ") <<
+    "overlap betwenn " << (smallerTableSorted ? "" : "not ") <<
+    "sorted smaller table, with " <<
+    size_tOrSize_tVectorToString(smallerTableAmountRows) <<
+    " rows and " << size_tOrSize_tVectorToString(smallerTableAmountColumns)
+    << " columns, and " << (biggerTableSorted ? "" : "not ") <<
+    "sorted bigger table, with " <<
+    size_tOrSize_tVectorToString(biggerTableAmountColumns) <<
+    " columns and " << size_tOrSize_tVectorToString(smallerTableAmountRows)
+    << " * " << size_tOrSize_tVectorToString(ratioRows) << " rows.";
+
+  // The lambdas for the join algorithms.
+  auto hashJoinLambda = makeHashJoinLambda();
+  auto joinLambda = makeJoinLambda();
+
+  // To reduce code duplication, the creation of the benchmark table is done
+  // per lambda.
+  auto createBenchmarkTable = [&tableDescriptor, &records](
+      const std::vector<size_t>& unconvertedRowNames){
+    // Creating the names for the rows for the benchmark table creation.
+    std::vector<std::string> rowNames(unconvertedRowNames.size());
+    std::ranges::transform(unconvertedRowNames, rowNames.begin(),
+        [](const size_t& number){return std::to_string(number);}, {});
+
+    records->addTable(tableDescriptor.str(), rowNames,
+        {"Merge/Galloping join", "Hash join"});
+  };
+
+  // Setup for easier creation of the tables, that will be joined.
+  IdTableAndJoinColumn smallerTable{makeIdTableFromVector({{}}), 0};
+  IdTableAndJoinColumn biggerTable{makeIdTableFromVector({{}}), 0};
+  auto replaceIdTables = [&overlap, &smallerTableSorted, &biggerTableSorted,
+       &smallerTable, &biggerTable](size_t smallerTableAmountRows,
+           size_t smallerTableAmountColumns, size_t ratioRows,
+           size_t biggerTableAmountColumns){
+         // Replacing the old id tables with newly generated ones, based
+         // on specification.
+         smallerTable.idTable = createRandomlyFilledIdTable(
+             smallerTableAmountRows, smallerTableAmountColumns, 0, 0, 500);
+         biggerTable.idTable = createRandomlyFilledIdTable(
+             smallerTableAmountRows * ratioRows, biggerTableAmountColumns,
+             0, 501, 1'000'000'000);
```

Comment:
A table should have a sample size (The number of possible values for entries) of `tableSize * ratioSampleSize`. Just adjust the arguments for the `createRandomlyFilledIdTable` here.

Oh, and sample size ratio should also be a parameter of makeBenchmarkTable for both tables. (It should be possible for it to be a vector.)

Reply:
Seems to work.